councilmatic_core/models.py

```diff
@@ -396,158 +584,46 @@ def updated_bills_since(cls, date_cutoff):
-        # load
-        return self.actions.order_by('-date', '-_bill__identifier', '-order')[:300]
-
-    @property
-    def recent_events(self):
-        # need to look up event participants by name
-        events = Event.objects.filter(participants__entity_type='organization', participants__entity_name=self.name)
-        events = events.order_by('-start_time').all()
-        return events
-
-    @property
-    def upcoming_events(self):
-        """
-        grabs events in the future
-        """
-        # need to look up event participants by name
-        events = Event.objects\
-                      .filter(participants__entity_type='organization', participants__entity_name=self.name)\
-                      .filter(start_time__gt=datetime.now(app_timezone))\
-                      .order_by('start_time')\
-                      .all()
-        return events
-
-    @property
-    def chairs(self):
-        if hasattr(settings, 'COMMITTEE_CHAIR_TITLE'):
-            return self.memberships.filter(role=settings.COMMITTEE_CHAIR_TITLE).filter(end_date__gt=datetime.now(app_timezone))
-        else:
-            return []
-
-    @property
-    def non_chair_members(self):
-        if hasattr(settings, 'COMMITTEE_MEMBER_TITLE'):
-            return self.memberships.filter(role=settings.COMMITTEE_MEMBER_TITLE).filter(end_date__gt=datetime.now(app_timezone))
-        else:
-            return []
-
-    @property
-    def all_members(self):
-        if hasattr(settings, 'COMMITTEE_MEMBER_TITLE'):
-            return self.memberships.filter(end_date__gt=datetime.now(app_timezone))
-        else:
-            return []
-
-    @property
-    def vice_chairs(self):
-        if hasattr(settings, 'COMMITTEE_VICE_CHAIR_TITLE'):
-            return self.memberships.filter(role=settings.COMMITTEE_VICE_CHAIR_TITLE).filter(end_date__gt=datetime.now(app_timezone))
-        else:
-            return []
-
-    @property
-    def link_html(self):
-        link_fmt = '<a href="{0}">{1}</a>'
-
-        if self.classification == 'committee':
-
-            try:
-                link_path = reverse('{}:committee_detail'.format(settings.APP_NAME), args=(self.slug,))
-            except NoReverseMatch:
-                link_path = reverse('committee_detail', args=(self.slug,))
-
-            return link_fmt.format(link_path, self.name)
-
-        if self.classification == 'legislature':
-
-            try:
-                link_path = reverse('{}:council_members'.format(settings.APP_NAME))
-            except NoReverseMatch:
-                link_path = reverse('council_members')
-
-            return link_fmt.format(link_path, self.name)
-
-        return self.name
+class BillSponsorship(opencivicdata.legislative.models.BillSponsorship):
+    class Meta:
+        proxy = True
 
+    bill = ProxyForeignKey(Bill, related_name='sponsorships', on_delete=models.CASCADE)
+    person = ProxyForeignKey(Person, null=True, on_delete=models.SET_NULL)
```

Comment:
interesting choice to make this nullable. in what instances are sponsorships not associated with a person? here are the sponsorships without associated people on my local machine:

```
| id                                   | name                       | entity_type | primary | classification | bill_id                                       | organization_id | person_id |
| ------------------------------------ | -------------------------- | ----------- | ------- | -------------- | --------------------------------------------- | --------------- | --------- |
| f0f9e1dc-108c-44b6-9209-78465bbfa023 | Lightfoot, Lori E.         | person      |    True | Primary        | ocd-bill/5b6563dd-f022-4584-9e87-4ef662575378 |                 |           |
| 3d36f500-083c-4cda-866a-31278d277e80 | Lightfoot, Lori E.         | person      |    True | Primary        | ocd-bill/4a712ecd-da48-43e8-8158-a1170bc5bec9 |                 |           |
| a7f9dc13-30f2-41ef-a581-fca2621c735b | La Spata, Daniel           | person      |    True | Primary        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 070f437a-26ee-4a14-b9cd-1e7de83fdc59 | Coleman, Stephanie D.      | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 4eea1d02-e6db-47d5-89b0-ca79f2eab604 | Taylor, Jeanette B.        | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| b90ca306-82a0-4aa1-8726-3f18a150c452 | Rodriguez, Michael D.      | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 6fa576fb-b99b-4025-9d88-bf929d0b7454 | Byron Sigcho-Lopez         | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 09bceabf-e54e-4e45-ba47-644ae6482e41 | Cardona, Jr., Felix        | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 055e6e86-0156-4f44-9ba3-48df89c94aac | Rodriguez Sanchez, Rossana | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 1d7367aa-a4da-4e54-ae11-c7997c31023a | Nugent, Samantha           | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| d9d44823-9f3f-4ecd-a1fa-07b50f0f8361 | Vasquez, Jr., Andres       | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 2f09b027-49bc-491a-a704-4dde4b82f628 | Gardiner, James M.         | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| eb07d993-757e-41b7-98e8-fa59df081f22 | Martin, Matthew J.         | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 70c011f1-ab3e-4742-9ef0-8ad42a6fbe01 | Hadden, Maria E.           | person      |   False | Regular        | ocd-bill/4bd88d59-e39e-468b-9643-5e1b59c23674 |                 |           |
| 8f8bcc77-bf07-480c-9833-1bf029e21a0d | Lightfoot, Lori E.         | person      |    True | Primary        | ocd-bill/0866399e-385d-4030-8158-fa4de682e614 |                 |           |
| 872a5b35-5e33-42db-9e07-e4cc23b8a489 | Lightfoot, Lori E.         | person      |    True | Primary        | ocd-bill/cecaf55f-b8a5-4fff-9221-330a73a27ad6 |                 |           |
| d97ced7c-728d-48a5-93d4-eb809945497b | Lightfoot, Lori E.         | person      |    True | Primary        | ocd-bill/3317b40f-9db1-4465-82cb-b70240177583 |                 |           |
| d6eea56a-7205-4f17-963e-8702a49bbec3 | Lightfoot, Lori E.         | person      |    True | Primary        | ocd-bill/56048709-7b05-4526-b270-5c6b0bf2d840 |                 |           |
| 0c8be0f5-770a-4b5b-bac8-91799e22b759 | Lightfoot, Lori E.         | person      |   False | Regular        | ocd-bill/e4548dee-9a08-4d62-a7e4-604358e4c6bf |                 |           |
| 4a11f867-be96-4eef-acce-b79406a0438a | Lightfoot, Lori E.         | person      |    True | Primary        | ocd-bill/21079f96-5361-4d6a-b7a4-61fa663746f9 |                 |           |
```

Reply:
An organization can be a sponsor.

Reply:
yes, that would be good.